x/axelarnet/keeper/msg_server.go

```diff
@@ -255,27 +255,33 @@ func (s msgServer) ExecutePendingTransfers(c context.Context, _ *types.ExecutePe
 	}
 
 	for _, pendingTransfer := range pendingTransfers {
-		recipient, err := sdk.AccAddressFromBech32(pendingTransfer.Recipient.Address)
-		if err != nil {
-			s.Logger(ctx).Error(fmt.Sprintf("discard invalid recipient %s and continue", pendingTransfer.Recipient.Address))
-			s.nexus.ArchivePendingTransfer(ctx, pendingTransfer)
-			continue
-		}
+		success := utils.RunCached(ctx, s, func(ctx sdk.Context) (bool, error) {
+			recipient, err := sdk.AccAddressFromBech32(pendingTransfer.Recipient.Address)
```

Comment:
Hmm, an address validation previously would be marked as archive but now as failed, which is inconsistent if we don't migrate the existing archived transfers to failed. Shall we continue marking these as archived so we don't make bad assumptions later? At the least there should be a warning comment here if we still strongly want to introduce this inconsistency

Reply:
It's a failed transfer and we can make logic cleaner with the newly introduced status. I think legacy shouldn't prevent us to improve and there is no third party relying on this.  sure can add comments